### PR TITLE
Implement searching products

### LINKED ISSUE #20: Add endpoint for searching

### LINKED ISSUE #21: Client side searching

### PR DESCRIPTION
**Pull request by songofroland**
resolves #20 
resolves #21

### CODE REVIEW

File: easywarehouse/templates/products/list.html [hunk 1]
```diff
@@ -2,17 +2,22 @@
...
-  <ul class="product-list">
-    {% for product in object_list %}
-      <li>
-        <a href="{{ product.get_absolute_url }}">
-          <img src="{{product.images.first.image.url}}" alt="{{product.name}} photo">
-          <div>{{ product.name }}</div>
-        </a>
-      </li>
-    {% endfor %}
-  </ul>
```
Comment (wilmhit) [inline easywarehouse/templates/products/list.html:17]: What is your HTML formatter?

File: easywarehouse/products/models.py [hunk 2]
```diff
@@ -30,6 +30,7 @@ class Product(models.Model):
     good_availability_threshold = models.PositiveIntegerField()
     tags = ArrayField(models.CharField(max_length=50), blank=True)
     description = models.TextField(default="")
+    main_image_url = models.URLField(null=True, blank=True)
```
[the remark this answers is not in the document]
Comment (wilmhit) [reply easywarehouse/products/models.py:33]: And there is no migration for this change

**Review by wilmhit — CHANGES_REQUESTED** [review]

File: easywarehouse/products/migrations/0006_auto_20210525_0739.py [hunk 3]
```diff
@@ -6,29 +6,29 @@
 class Migration(migrations.Migration):
 
     dependencies = [
-        ('products', '0005_auto_20210521_1325'),
```
[the remark this answers is not in the document]
Comment (songofroland) [reply easywarehouse/products/migrations/0006_auto_20210525_0739.py:9]: should be, I'll have a look

File: easywarehouse/products/models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (songofroland) [reply easywarehouse/products/models.py:33]: > And there is no migration for this change

Good catch, I'll add it

> This should be a relation to an image. Image class stores urls.
Actually none of the classes stores URL now, They are resolved dynamically. 

I found out that the easiest way is to manually enter the URL per each product. Also, this is somehow a config value - it tells that the particulat image is the main image of the product. If this does not convince you, let's make a call

File: easywarehouse/templates/products/list.html (hunk 1 above)
Comment (songofroland) [reply easywarehouse/templates/products/list.html:17]: Pycharm's default, we can agree on one style though

File: logstash/logstash.conf [hunk 4]
```diff
@@ -5,6 +5,7 @@ input {
         jdbc_user => "${LOGSTASH_JDBC_USERNAME}"
         jdbc_password => "${LOGSTASH_JDBC_PASSWORD}"
         schedule => "* * * * *"
+        # TODO: Join products with categories
```
[the remark this answers is not in the document]
Comment (songofroland) [reply logstash/logstash.conf:8]: Yeah, I'm not keen to invest more time this week. I'm pretty sure there will be some testing made, and we'll check that searching on categories does not work, so I'd park it for now

**Review by wilmhit — APPROVED** [review]

**Pull request merged**